test/videojs.ads.snapshot.js

```diff
@@ -1,467 +1,365 @@
-  player.ads.startLinearAdMode();
-  player.ads.endLinearAdMode();
-
-  ok(!srcModified, 'the src was reset');
-  ok(playCalled, 'content playback resumed');
-
-  player.trigger('playing');
-  equal(contentPlaybackFired, 1, 'A content-playback event should have triggered');
-  equal(contentPlaybackReason, 'playing', 'The reason for content-playback should have been playing');
+  this.player.ads.startLinearAdMode();
+  this.player.ads.endLinearAdMode();
+  assert.ok(playSpy.called, 'content playback resumed');
+  assert.ok(srcSpy.alwaysCalledWithExactly(), 'the src was reset');
+  this.player.trigger('playing');
+  assert.ok(this.contentPlaybackSpy.calledOnce, 'A content-playback event should have triggered');
+  assert.strictEqual(this.contentPlaybackReason(), 'playing', 'The reason for content-playback should have been playing');
 
   // the src wasn't changed, so we shouldn't be waiting on loadedmetadata to
   // update the currentTime
-  player.trigger('loadedmetadata');
-  ok(!currentTimeModified, 'no seeking occurred');
+  this.player.trigger('loadedmetadata');
+  assert.ok(currentTimeSpy.alwaysCalledWithExactly(), 'no seeking occurred');
 });
 
-test('snapshot does not resume playback after post-rolls', function() {
-  var playCalled = false;
+QUnit.test('snapshot does not resume playback after post-rolls', function(assert) {
+  assert.expect(7);
 
   // start playback
-  player.src('http://media.w3.org/2010/05/sintel/trailer.mp4');
-  player.trigger('loadstart');
-  player.trigger('loadedmetadata');
-  player.trigger('adsready');
-  player.trigger('play');
-
-  // spy on relevant player methods
-  player.play = function() {
-    playCalled = true;
-  };
-
-  //trigger an ad
-  player.ads.startLinearAdMode();
-  player.src('//example.com/ad.mp4');
-  player.trigger('loadstart');
-  player.trigger('loadedmetadata');
-  player.ads.endLinearAdMode();
-
-  //resume playback
-  player.src('http://media.w3.org/2010/05/sintel/trailer.mp4');
-  player.trigger('loadstart');
-  player.trigger('canplay');
-  ok(playCalled, 'content playback resumed');
-  player.trigger('playing');
-  equal(contentPlaybackFired, 1, 'A content-playback event should have been triggered');
-  equal(contentPlaybackReason, 'playing', 'The reason for content-playback should have been playing');
+  this.player.src('http://media.w3.org/2010/05/sintel/trailer.mp4');
+  this.player.trigger('loadstart');
+  this.player.trigger('loadedmetadata');
+  this.player.trigger('adsready');
+  this.player.tech.trigger('play');
+  var playSpy = sinon.spy(this.player, 'play');
+
+  // trigger an ad
+  this.player.ads.startLinearAdMode();
+  this.player.src('//example.com/ad.mp4');
+  this.player.trigger('loadstart');
+  this.player.trigger('loadedmetadata');
+  this.player.ads.endLinearAdMode();
+
+  // resume playback
+  this.player.src('http://media.w3.org/2010/05/sintel/trailer.mp4');
+  this.player.trigger('loadstart');
+  this.player.trigger('canplay');
+  this.clock.tick(1000);
```

Comment:
I'm adding a comment, but it's necessary because the "canplay" event causes the `restorePlayerSnapshot` to be called which queues up 20 timeouts at 50ms each (hence the 1000ms).